### DISCUSSION TITLE
Unsure how to use the foreign feature

### DISCUSSION THREAD

**Discussion by tfc** [purs-nix/purs-nix#28] [Q&A]
Hi,

i created an example repo that shows the question/situation that i have with the [foreign feature (docs)](https://github.com/purs-nix/purs-nix/blob/master/docs/foreign.md).

https://github.com/tfc/purescript-recharts

So i got the bundling running using npmlock2nix etc. and this is all fine: `nix run github:tfc/purescript-recharts` works nicely.

But, am i really supposed [to do this (link to the flake.nix line)](https://github.com/tfc/purescript-recharts/blob/main/flake.nix#L55) for all my project's transitive dependencies:

```nix
# flake.nix content

# ... purs call...
            foreign."React.Basic".node_modules = nodeModules;
            foreign."React.Basic.DOM".node_modules = nodeModules;
            foreign."React.Basic.DOM.Client".node_modules = nodeModules;
            foreign."React.Basic.DOM.Internal".node_modules = nodeModules;
            foreign."Recharts".node_modules = nodeModules;
# ...
```

?

**Comment by ursi** [comment]
In an ideal world, all the packages would declare their foreign dependencies so you don't have to do it yourself, but unfortunately, not a single package in the official package set is using this API at the time of writing, so you're stuck having to do this if you want to make a fully self-contained package.

A tip: you can automate away some of this boilerplate using `lib.genAttrs` with a list of module names.

**Comment by tfc** [comment]
I see. Thank you.

**Comment by ursi** [comment]
Maybe at some point there will be an official or unofficial collection of foreign dependencies for popular packages.
